File: app/src/main/java/com/a60n1/ejashkojme/service/MessagingService.java
package com.a60n1.ejashkojme.service;

import android.app.NotificationManager;
import android.app.PendingIntent;
import android.content.Intent;
import android.graphics.Bitmap;
import android.graphics.BitmapFactory;

import androidx.annotation.NonNull;
import androidx.core.app.NotificationCompat;

import com.a60n1.ejashkojme.R;
import com.google.firebase.messaging.RemoteMessage;

import java.util.Objects;

public class MessagingService extends com.google.firebase.messaging.FirebaseMessagingService {
    @Override
    public void onMessageReceived(@NonNull RemoteMessage remoteMessage) {
        super.onMessageReceived(remoteMessage);

        String title = Objects.requireNonNull(remoteMessage.getNotification()).getTitle();
        String body = remoteMessage.getNotification().getBody();

        String click_action = remoteMessage.getNotification().getClickAction();

        String from_user_id = remoteMessage.getData().get("from_user_id");
        String address_user_id = remoteMessage.getData().get("address_user_id");

        Bitmap icon = BitmapFactory.decodeResource(getResources(), R.mipmap.ic_launcher_round);
        NotificationCompat.Builder mBuilder =
                new NotificationCompat.Builder(this)
                        .setSmallIcon(R.mipmap.ic_launcher_round)
                        .setLargeIcon(icon)
                        .setContentTitle(title)
                        .setContentText(body);
        Intent resultIntent = new Intent(click_action);
        resultIntent.putExtra("user_id", from_user_id);
        resultIntent.putExtra("origin", address_user_id);

        PendingIntent resultPendingIntent =
                PendingIntent.getActivity(
                        this,
                        0,
                        resultIntent,
                        PendingIntent.FLAG_UPDATE_CURRENT
                );
        mBuilder.setContentIntent(resultPendingIntent);
        int mNotificationId = (int) System.currentTimeMillis();

        NotificationManager notificationManager =
                (NotificationManager) getSystemService(NOTIFICATION_SERVICE);
        Objects.requireNonNull(notificationManager).notify(mNotificationId, mBuilder.build());
    }
}

File: app/src/main/java/com/a60n1/ejashkojme/service/MessagingService.kt
package com.a60n1.ejashkojme.service

import android.app.NotificationManager
import android.app.PendingIntent
import android.content.Context
import android.content.Intent
import android.graphics.BitmapFactory
import androidx.core.app.NotificationCompat
import com.a60n1.ejashkojme.R
import com.google.firebase.messaging.FirebaseMessagingService
import com.google.firebase.messaging.RemoteMessage
import java.util.*

@Suppress("DEPRECATION")
class MessagingService : FirebaseMessagingService() {
    override fun onMessageReceived(remoteMessage: RemoteMessage) {
        super.onMessageReceived(remoteMessage)
        val title = remoteMessage.notification!!.title
        val body = remoteMessage.notification!!.body
        val clickAction = remoteMessage.notification!!.clickAction
        val fromUserId = remoteMessage.data["from_user_id"]
        val addressUserId = remoteMessage.data["address_user_id"]
        val icon = BitmapFactory.decodeResource(resources, R.mipmap.ic_launcher_round)
        val mBuilder = NotificationCompat.Builder(this)
                .setSmallIcon(R.mipmap.ic_launcher_round)
                .setLargeIcon(icon)
                .setContentTitle(title)
                .setContentText(body)
        val resultIntent = Intent(clickAction)
        resultIntent.putExtra("user_id", fromUserId)
        resultIntent.putExtra("origin", addressUserId)
        val resultPendingIntent = PendingIntent.getActivity(
                this,
                0,
                resultIntent,
                PendingIntent.FLAG_UPDATE_CURRENT
        )
        mBuilder.setContentIntent(resultPendingIntent)
        val mNotificationId = System.currentTimeMillis().toInt()
        val notificationManager = getSystemService(Context.NOTIFICATION_SERVICE) as NotificationManager
        Objects.requireNonNull(notificationManager).notify(mNotificationId, mBuilder.build())
    }
}